### ISSUE TITLE
Runtime Type Usage Unnecessary Generic Arguments Rule

### ISSUE THREAD

**Issue by hmc-cs-mdrissi** [microsoft/pyright#4026]
**Describe the bug**
```python
print(list) # Good No Error
x: list # Good Error 
print(list[dict]) # Unexpected error: Expected type arguments for generic class "dict"
y: list[dict] # Good Error
```

The main unexpected line is `print(list[dict])` as list[dict] here is not in an annotation context.

**Expected behavior**
I would expect runtime usage of a type in a non annotation context to not require generic arguments. That rule looks to be followed top level, but not with nested generics.

**VS Code extension or command-line**
Noticed first from pylance and confirmed 1.1.274 pyright produces same error message.

**Comment by erictraut** [comment]
In the expression `list[dict]`, you're using `dict` to specialize the generic class `list`, but you are specializing it with a type that is itself generic and is not specialized. I think it's reasonable in this case to emit a `reportMissingTypeArguments` diagnostic.

The `reportMissingTypeArguments` check is not enabled by default in "basic" type checking mode. It's a strict-mode check, so I think it's OK for it to be a bit more opinionated.
